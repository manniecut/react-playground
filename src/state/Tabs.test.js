import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Tabs from './Tabs';

describe(`Tabs Component`, () => {
    const tabsProp = [
        {
            name: 'First tab',
            content: 'Loruem imspon fadolra sit amet csonecut dipipti fjkljoal.'
        },
        {
            name: 'Second tab',
            content: 'Laboriosam execritanonetmiom quos cosnceuter elalelflewk.'
        },
        {
            name: '3rd tab',
            content: 'Fugit, sapiente aspernatur corpiris velit, caclkjeoifasl.'
        }
    ];
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    /*it('renders the first tab by default', () => {
        const tree = renderer.create(<Tabs tabs={tabsProp} />)
        expect(tree).toMatchSnapshot()
    })*/
    it('renders empty when given no tabs', () => {
        const wrapper = shallow(<Tabs />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders the first tab by default', () => {
        const wrapper = shallow(<Tabs tabs={tabsProp} />);
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Tabs tabs={tabsProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})