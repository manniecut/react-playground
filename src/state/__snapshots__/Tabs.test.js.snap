// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closes the first tab and opens any clicked tab 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    3rd tab
  </button>
  <div
    className="content"
  >
    Laboriosam execritanonetmiom quos cosnceuter elalelflewk.
  </div>
</div>
`;

exports[`Tabs Component renders empty when given no tabs 1`] = `<div />`;

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    3rd tab
  </button>
  <div
    className="content"
  >
    Loruem imspon fadolra sit amet csonecut dipipti fjkljoal.
  </div>
</div>
`;
